sql/hive-thriftserver/v1.2.1/src/main/java/org/apache/hive/service/cli/ColumnValue.java

```diff
@@ -195,7 +195,7 @@ public static TColumnValue toTColumnValue(Type type, Object value) {
     case INTERVAL_DAY_TIME_TYPE:
       return stringValue((HiveIntervalDayTime) value);
     case DECIMAL_TYPE:
-      return stringValue(((HiveDecimal)value));
+      return getDecimalType(value);
```

Comment:
```return stringValue(((BigDecimal)value).toPlainString());```?


https://github.com/apache/hive/blob/release-2.3.5-rc0/storage-api/src/java/org/apache/hadoop/hive/common/type/HiveDecimal.java#L725

Reply:
We also need to update v2.3.5's `ColumnValue`: https://github.com/apache/spark/blob/master/sql/hive-thriftserver/v2.3.5/src/main/java/org/apache/hive/service/cli/ColumnValue.java#L201-L202

Reply:
> `return stringValue(((BigDecimal)value).toPlainString());`?
> 
> https://github.com/apache/hive/blob/release-2.3.5-rc0/storage-api/src/java/org/apache/hadoop/hive/common/type/HiveDecimal.java#L725

you mean getDecimalType() function is not required?

Reply:
Yes.

Reply:
k will update  just need one clarification 'toPlainString()' conversion takes care necessariy normalizations as all? because create function does that

![image](https://user-images.githubusercontent.com/12999161/62694050-38a36500-b9f1-11e9-9853-9b1b755c755d.png)

Reply:
ultimately it converts to HiveDecimal and then we are sending back the string value.

Reply:
Please avoid using `HiveDecimal` because it may cause inconsistent results for `ColumnBasedSet` implementations and `RowBasedSet` implementations:
https://github.com/apache/spark/blob/02c33694c8254f69cb36c71c0876194dccdbc014/sql/hive-thriftserver/v1.2.1/src/main/java/org/apache/hive/service/cli/RowSetFactory.java#L35-L40

Reply:
i didn't got exactly pardon me , but if client expects HiveDecimal then?